### DISCUSSION TITLE
[Microsoft.Extensions.Http.Resilience] Why is some classes/methods `Experimental` - but extension methods utilizing these are not?

### DISCUSSION THREAD

**Discussion by martinhansen** [dotnet/extensions#5204] [Q&A]
I am trying to use this library for resilience with http clients, and when looking through the source code, i find that some classes and methods are marked with the `Experimantal` attribute, while extension methods utilizing these methods and/or classes are not.

Take for instance [ResilienceHandler](https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandler.cs) or [HttpClientResiliencePredicates](https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.Http.Resilience/Polly/HttpClientResiliencePredicates.cs) which is marked as (or has methods marked with) `Experimental` but utilized in [ResilienceHttpClientBuilderExtensions.Resilience.cs](https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHttpClientBuilderExtensions.Resilience.cs) - which is not marked as `Experimental`.

I need some additional features, so I need to initialize the `ResilienceHandler.cs` myself - but is it safe to use? If going by the documentation, it's not, as it is experimental. But using the extension methods is?

Is it a mistake that (some) of these classes/methods is still marked as `Experimental`, and if not, why is the extension methods not also marked with `Experimental`?

Basically I'm trying to figure out whether or not `ResilienceHandler.cs` and other experimental classes/methods in this library is safe to use for production environments or not.

I'm not afraid of it changing between versions without notice. I can handle that. I'm just worried about the safety of use in production

Hope you can help me with some insights.

**Comment by RussKie** [comment]
That's a good question :)

@geeknoid @iliar-turdushev any recollections of why this may have happened or thoughts?

**Comment by martintmk** [comment]
I think it's just general guideline we used to introduce any new API as experimental :).

I don't see any problems with dropping it for `ResilienceHandler`.
